File: server-js/server/output.js
// Print common output data.
const outputHeader = (address) => {
  const datetime = new Date().toISOString().replace('T', ' ').substring(0, 19).replace(/-/g, '/');
  process.stdout.write(`\n${datetime.padEnd(20, ' ')} ${address.padEnd(15, ' ')} > `);
}

// Print server initialization data.
export const outputInit = () => {
  outputHeader('0.0.0.0');
  process.stdout.write('INIT\n');
}

// Print received request data.
export const outputRequest = (request) => {
  const url = request.url.split('?');
  outputHeader(request.socket.remoteAddress.slice(7));
  process.stdout.write(`${request.method.padEnd(6, ' ')} ${url[0]}\n`);
  if (url[1]) {
    process.stdout.write(`${'>'.padStart(38, ' ')} ${'QUERY'.padEnd(6, ' ')} ${url[1]}\n`);
  }
}

File: server-c/server/output.c
#include "./output.h"

// Print common output data.
static void outputHeader(char* address) {
  char datetime[20];
  time_t _datetime = time(NULL);
  strftime(datetime, sizeof(datetime), "%Y/%m/%d %H:%M:%S", localtime(&_datetime));
  printf("\n%-20s %-15s > ", datetime, address);
}

// Print server initialization data.
void outputInit() {
  outputHeader("0.0.0.0");
  printf("INIT\n");
}

// Print received request data.
void outputRequest(struct ls_connection* socket, struct ls_http_request request) {
  outputHeader(inet_ntoa(socket->address.sin_addr));
  printf("%-6s %s\n", request.method, request.uri);
  if (request.query) {
    printf("%38s %-6s %s\n", ">", "QUERY", request.query);
  }
}